src/federation/index.ts

```diff
@@ -904,6 +906,61 @@ federation.setObjectDispatcher(
   },
 );
 
+federation.setObjectDispatcher(
+  Flag,
+  "/reports/{id}",
+  async (ctx, { id }) => {
+    const report = await db.query.reports.findFirst({
+      where: eq(reports.id, id),
+      with: {
+        account: {
+          columns: { iri: true }
+        },
+        targetAccount: {
+          columns: {
+            iri: true
+          }
+        }
+      },
+    });
```

Comment:
The SQL query this generates is a bit of a mouthful: 

```sql
select "reports"."id",
  "reports"."iri",
  "reports"."account_id",
  "reports"."target_account_id",
  "reports"."created",
  "reports"."comment",
  "reports"."posts",
  "reports_account"."data" as "account",
  "reports_targetAccount"."data" as "targetAccount"
from "reports"
  left join lateral (
    select json_build_array("reports_account"."iri") as "data"
    from (
        select *
        from "accounts" "reports_account"
        where "reports_account"."id" = "reports"."account_id"
        limit 1
      ) "reports_account"
  ) "reports_account" on true
  left join lateral (
    select json_build_array("reports_targetAccount"."iri") as "data"
    from (
        select *
        from "accounts" "reports_targetAccount"
        where "reports_targetAccount"."id" = "reports"."target_account_id"
        limit 1
      ) "reports_targetAccount"
  ) "reports_targetAccount" on true
where "reports"."id" = '0192b649-9c5a-714a-ae0f-e9016fbad48f'
limit 1
```

But we only need back the `iri` for both accounts, not the full account object. Arguably I could write something more optimised with a direct `db.select` type query.

Reply:
I found it okay as it wouldn't be requested so much.